src/System.Drawing.Common/tests/mono/System.Imaging/MetafileTest.cs

```diff
@@ -0,0 +1,576 @@
+        public void GetMetafileHeader_WmfPlaceable()
+        {
+            using (Metafile mf = new Metafile(Helpers.GetTestBitmapPath(WmfPlaceable)))
+            {
+                MetafileHeader header1 = mf.GetMetafileHeader();
+                Check_MetafileHeader_WmfPlaceable(header1);
+
+                MetafileHeader header2 = mf.GetMetafileHeader();
+                Assert.False(Object.ReferenceEquals(header1, header2));
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void GetMetafileHeader_FromFile_WmfPlaceable()
+        {
+            using (Metafile mf = new Metafile(Helpers.GetTestBitmapPath(WmfPlaceable)))
+            {
+                MetafileHeader header1 = mf.GetMetafileHeader();
+                Check_MetafileHeader_WmfPlaceable(header1);
+
+                MetaHeader mh1 = header1.WmfHeader;
+                Check_MetaHeader_WmfPlaceable(mh1);
+
+                MetaHeader mh2 = mf.GetMetafileHeader().WmfHeader;
+                Assert.False(Object.ReferenceEquals(mh1, mh2));
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void GetMetafileHeader_FromFileStream_WmfPlaceable()
+        {
+            using (FileStream fs = File.OpenRead(Helpers.GetTestBitmapPath(WmfPlaceable)))
+            using (Metafile mf = new Metafile(fs))
+            {
+                MetafileHeader header1 = mf.GetMetafileHeader();
+                Check_MetafileHeader_WmfPlaceable(header1);
+
+                MetaHeader mh1 = header1.WmfHeader;
+                Check_MetaHeader_WmfPlaceable(mh1);
+
+                MetaHeader mh2 = mf.GetMetafileHeader().WmfHeader;
+                Assert.False(Object.ReferenceEquals(mh1, mh2));
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void GetMetafileHeader_FromMemoryStream_WmfPlaceable()
+        {
+            string filename = Helpers.GetTestBitmapPath(WmfPlaceable);
+            using (MemoryStream ms = new MemoryStream(File.ReadAllBytes(filename)))
+            using (Metafile mf = new Metafile(ms))
+            {
+                MetafileHeader header1 = mf.GetMetafileHeader();
+                Check_MetafileHeader_WmfPlaceable(header1);
+
+                MetaHeader mh1 = header1.WmfHeader;
+                Check_MetaHeader_WmfPlaceable(mh1);
+
+                MetaHeader mh2 = mf.GetMetafileHeader().WmfHeader;
+                Assert.False(Object.ReferenceEquals(mh1, mh2));
+            }
+        }
+
+        public static void Check_MetafileHeader_Emf(MetafileHeader header)
+        {
+            Assert.Equal(MetafileType.Emf, header.Type);
+            Assert.Equal(65536, header.Version);
+            // extactly the filesize
+            Assert.Equal(20456, header.MetafileSize);
+
+            Assert.Equal(0, header.Bounds.X);
+            Assert.Equal(0, header.Bounds.Y);
+
+            try
+            {
```

Comment:
This whole try, catch block and be just simplified to: `Assert.Throws<ArgumentException>(() => header.WmfHeader);`